crates/compiler/mono/src/tail_recursion.rs

```diff
@@ -931,32 +1051,104 @@ impl<'a> TrmcEnv<'a> {
                                 Vec::from_iter_in(call.arguments.iter().copied(), env.arena);
                             jump_arguments.push(new_hole_symbol);
                             jump_arguments.push(self.head_symbol);
-
                             let jump =
                                 Stmt::Jump(self.joinpoint_id, jump_arguments.into_bump_slice());
 
-                            let output = let_tag_arg_null(arena.alloc(
-                                //
-                                let_tag(arena.alloc(
-                                    //
-                                    let_new_hole(arena.alloc(
+                            match call_location {
```

Comment:
this nesting is getting  pretty severe, maybe this logic can be factored out into its own function?